.circleci/config.yml

```diff
@@ -1,65 +1,112 @@
+  cypress: cypress-io/cypress@1
 jobs:
   build:
     docker:
-      - image: docker:17.12.0-ce-git
-    working_directory: /usr/src
+      - image: circleci/node:lts-browsers
+    working_directory: ~/repo
     steps:
       - checkout
-      - setup_remote_docker
-      - run:
-          name: Execute Pipeline (Build Source -> Test -> Build Web Server)
-          # code is checked out in the pipeline, consider copying from CI to Pipeline
-          # consider caching node_modules or switching to npm ci option when it becomes stable
-          command: |
-            docker build -f integration.Dockerfile . -t lemon-mart:$CIRCLE_BRANCH --build-arg CACHEBUST=$(date +%s)
-            mkdir -p docker-cache
-            docker save -o docker-cache/built-image.tar lemon-mart:$CIRCLE_BRANCH
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+      - run: npm ci
+
+      # force update the webdriver, so it runs with latest version of Chrome
+      - run: cd ./node_modules/protractor && npm i webdriver-manager@latest
+
       - save_cache:
-          key: built-image-{{ .BuildNum }}
           paths:
-            - docker-cache
-      - store_artifacts:
-          path: docker-cache/built-image.tar
-          destination: built-image.tar
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+
+      - run: npm run style
+      - run: npm run lint
+      - run: npm run build:prod
+      - run: npm test -- --watch=false --code-coverage
+      - run: npm run e2e
+
       - store_test_results:
-          path: tests/units.html
+          path: ./test_results
+      - store_artifacts:
+          path: ./coverage
+
+      # upload coverage report to coveralls for readme badge support
+      # requires config.yml version 2.1 and orbs: coveralls: coveralls/coveralls@1.0.4
+      # requires coveralls account and token named COVERALLS_REPO_TOKEN specific to repo to be stored in CircleCI
+      - coveralls/upload
+
+      - run:
+          name: Tar & Gzip compiled app
+          command: tar zcf dist.tar.gz dist/lemon-mart
+      - store_artifacts:
+          path: dist.tar.gz
 
-  publish:
+      - run:
+          name: Move compiled app to workspace
+          command: |
+            set -exu
+            mkdir -p /tmp/workspace/dist
+            mv dist/lemon-mart /tmp/workspace/dist/
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - dist/lemon-mart
+  deploy:
     docker:
-      - image: docker:17.12.0-ce-git
-    working_directory: /usr/src
+      - image: circleci/node:lts
+    working_directory: ~/repo
     steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: npx now --token $NOW_TOKEN --platform-version 2 --prod --name lemon-mart /tmp/workspace/dist/lemon-mart
+  cypress/run:
```

Comment:
since you are using the orb here in order to avoid surprises, just do what the example shows https://github.com/cypress-io/circleci-orb#record-on-dashboard and no other commands are necessary

```
version: 2.1
orbs:
  cypress: cypress-io/cypress@1
workflows:
  build:
    jobs:
      - cypress/run:
          record: true
```
it just seems to me that by using job name "cypress/run" that comes from Orb "cypress" you import, but then writing your own steps Circle is all mixing it together